homeassistant/data_entry_flow.py

```diff
@@ -198,6 +199,23 @@ def async_progress_by_handler(
             self._async_progress_by_handler(handler), include_uninitialized
         )
 
+    @callback
+    def async_progress_by_init_data_type(
+        self,
+        init_data_type: type,
+        matcher: Callable[[Any], bool],
```

Comment:
I made this take a callable as I didn't want to expose the underlying `FlowHandler` to this API so the implementation can change later if needed.